### PR TITLE
Data from CMS instead of from the repo itself

### PR DESCRIPTION
**Pull request by KMontag42**
lots of stuff in here:

- all data about artefacts, classes, factions, and characters comes from the [simple json cms](https://simplejsoncms.com/api/mdb18slfe7)
- remove a lot of code related to managing said data in the repo
- remove all of the images as they are hosted on imgur now :)
- some mild refactoring of the builder component to make things less complicated
- renaming spells to artefacts for consistency

### CODE REVIEW

File: components/Builder.tsx [hunk 1]
```diff
@@ -69,117 +42,59 @@ export default function Builder() {
...
+
+  const charactersNotInFormation = Object.values(Characters).filter(
+    (character) => !formation.includes(character.name),
+  );
+  const charactersInFormation = formation.map((x) => {
+    if (x === "" || x === undefined) {
+      return undefined;
+    }
+    return Characters[x.toLowerCase()];
+  });
```
Comment (DMills94) [inline components/Builder.tsx:68]: Might make more sense to chain a filter into a map here so we're not return an array with undefined valeus
```
formation.filter(x=>x).map(x=>blah)
```

File: components/Providers.tsx [hunk 2]
```diff
@@ -1,16 +1,9 @@
-'use client';
+import React from "react";
```
Comment (DMills94) [inline components/Providers.tsx:1]: do we need this line?

File: .env [hunk 3]
```diff
@@ -0,0 +1 @@
+NEXT_PUBLIC_SIMPLEJSONCMS_ID=mdb18slfe7
```
Comment (DMills94) [inline .env:1]: i guess without the password nobody can do anything with this?

**Review by DMills94 — DISMISSED** [review]

File: .env (hunk 3 above)
Comment (KMontag42) [reply .env:1]: yeah the password is what lets people modify the CMS, this is just used for GET requests

File: components/Providers.tsx (hunk 2 above)
Comment (KMontag42) [reply components/Providers.tsx:1]: we do not :jjnod:

File: components/Builder.tsx (hunk 1 above)
Comment (KMontag42) [reply components/Builder.tsx:68]: we need the empty spaces in the array to ensure characters are in the right numbered slot

File: .trunk/trunk.yaml
Comment (SwiftieHusband) [inline .trunk/trunk.yaml:1]: What is trunk? This is a versioning thing right?

File: components/Builder.tsx (hunk 1 above)
Comment (SwiftieHusband) [reply components/Builder.tsx:68]: Honestly, reworking the formation as a whole to be a 2D Array would be ideal here. And everything should be ID based, not string based. But that's a bigger rework and for the future.

```
"Grid": [
                [1, 1, 1, 0],
                [1, 1, 1, 1],
                [1, 1, 1, 1],
                [0, 0, 1, 1]
            ]
```

1 is an available space, 0 is not. And then when you drop a hero into the grid, that 1 would change to being an ID of the hero, so say Reinier is 51, that entry would become 51.

**Review by SwiftieHusband — APPROVED** [review]

File: .trunk/trunk.yaml
Comment (KMontag42) [reply .trunk/trunk.yaml:1]: Trunk is a meta-linter. 

https://docs.trunk.io/check

File: components/Builder.tsx (hunk 1 above)
Comment (KMontag42) [reply components/Builder.tsx:68]: > Honestly, reworking the formation as a whole to be a 2D Array would be ideal here. And everything should be ID based, not string based. But that's a bigger rework and for the future.
> 
> ```
> "Grid": [
>                 [1, 1, 1, 0],
>                 [1, 1, 1, 1],
>                 [1, 1, 1, 1],
>                 [0, 0, 1, 1]
>             ]
> ```
> 
> 1 is an available space, 0 is not. And then when you drop a hero into the grid, that 1 would change to being an ID of the hero, so say Reinier is 51, that entry would become 51.

I’m going to make this change as a fast follow to this PR landing

**Pull request merged**